libs/wire-api-federation/src/Wire/API/Federation/Endpoint.hs

```diff
@@ -90,3 +104,104 @@ type instance
 type instance
   MkNotificationFedEndpoint m s ('Just v) p =
     NotificationFedEndpointWithMods m (Versioned v s) s p
+
+type OriginIpHeaderName = "Wire-Origin-IP" :: Symbol
+
+-- | The remote backend's origin IP is best-effort forensic metadata only.
+--   Do not use for auth, policy, or attribution; cert identity is authoritative.
+--   IP reflects only the socket peer at our edge (often LB/NAT/egress) and may be inaccurate.
+data OriginIpHeader
+
+instance (RoutesToPaths api) => RoutesToPaths (OriginIpHeader :> api) where
+  getRoutes = getRoutes @api
+
+type instance SpecialiseToVersion v (OriginIpHeader :> api) = OriginIpHeader :> SpecialiseToVersion v api
```

Comment:
I think it is not needed

Reply:
I guess it's not needed because `SpecialiseToVersion` isn't used for the federation API? Doesn't hurt to keep it though, I'd say.